apis/infrastructure/v1beta1/bootstrapkubeconfig_webhook_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1_test
+
+import (
+	b64 "encoding/base64"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	byohv1beta1 "github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/apis/infrastructure/v1beta1"
+	"github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/test/builder"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api/util/patch"
+)
+
+var _ = Describe("BootstrapKubeconfig Webhook", func() {
+
+	var (
+		bootstrapKubeconfig         *byohv1beta1.BootstrapKubeconfig
+		err                         error
+		defaultNamespace            = "default"
+		testBootstrapKubeconfigName = "test-bootstrap-kubeconfig"
+		testServerEmpty             = ""
+		testServerWithoutScheme     = "abc.com"
+		testServerWithoutHostname   = "https://test-server"
+		testServerWithoutPort       = "https://test.com"
+		testServerValid             = "https://abc.com:1234"
+		testCADataEmpty             = ""
+		testCADataInvalid           = "test-ca-data"
+		testPEMDataInvalid          = b64.StdEncoding.EncodeToString([]byte(testCADataInvalid))
+	)
+	Context("When BootstrapKubeconfig gets a create request", func() {
+
+		It("should reject the request if APIServer field is empty", func() {
```

Comment:
Can add a test case if `url.Parse(r.Spec.APIServer)` returns an error

Reply:
Well, was looking for it. Turns out, [url.Parse](https://pkg.go.dev/net/url#Parse) returns empty or the characters as is in most cases - even for absurd strings. It is apparently due to "parsing ambiguities". I don't entirely follow what that means. But happy to add a case if I can find a string that causes `url.Parse` to return error

Reply:
AFAIR, if you add space in protocol then it will give you an error eg. `htt p://test.com`

Reply:
Ah! amazing. Adding it now :)